fs/nfsd/nfs41acl_xdr.h

```diff
@@ -0,0 +1,61 @@
+#ifndef _NFS41ACL_H_RPCGEN
+#define _NFS41ACL_H_RPCGEN
+
+#define NA41_NAME "system.nfs4_acl_xdr"
+
+typedef u_int acetype4;
+#define ACE4_FILE_INHERIT_ACE 0x00000001
+#define ACE4_DIRECTORY_INHERIT_ACE 0x00000002
+#define ACE4_NO_PROPAGATE_INHERIT_ACE 0x00000004
+#define ACE4_INHERIT_ONLY_ACE 0x00000008
+#define ACE4_SUCCESSFUL_ACCESS_ACE_FLAG 0x00000010
+#define ACE4_FAILED_ACCESS_ACE_FLAG 0x00000020
+#define ACE4_IDENTIFIER_GROUP 0x00000040
+#define ACE4_INHERITED_ACE 0x00000080
+#define NFS41_FLAGS	(ACE4_DIRECTORY_INHERIT_ACE| \
+			 ACE4_FILE_INHERIT_ACE| \
+			 ACE4_NO_PROPAGATE_INHERIT_ACE| \
+			 ACE4_INHERIT_ONLY_ACE| \
+			 ACE4_INHERITED_ACE| \
+			 ACE4_IDENTIFIER_GROUP)
+
+typedef u_int aceflag4;
+#define ACEI4_SPECIAL_WHO 0x00000001
+
+typedef u_int aceiflag4;
+#define ACE4_SPECIAL_OWNER 1
+#define ACE4_SPECIAL_GROUP 2
+#define ACE4_SPECIAL_EVERYONE 3
+
+typedef u_int acemask4;
+
+struct nfsace4i {
+	acetype4 type;
+	aceflag4 flag;
+	aceiflag4 iflag;
+	acemask4 access_mask;
+	u_int who;
```

Comment:
I wonder why is this u_int, not u32 or something?  You later have to explicitly convert to u_int and back and assert overflows during conversions.